### PR TITLE
[XNNPACK] optimize the compute performance

### PR DESCRIPTION
**Pull request by huningxin**
Only call xnn_setup_runtime when any external pointers change and use unordered_map for inputs and outputs.

@fujunwei @Honry , PTAL. Thanks!

### CODE REVIEW

File: src/webnn_native/xnnpack/GraphXNN.cpp [hunk 1]
```diff
@@ -743,30 +750,43 @@ namespace webnn_native::xnnpack {
...
 
     MaybeError Graph::ComputeImpl(NamedInputsBase* inputs, NamedOutputsBase* outputs) {
-        std::vector<xnn_external_value> externalValues;
-        for (auto& input : inputs->GetRecords()) {
-            if (mExternals.find(input.first) == mExternals.end()) {
-                return DAWN_VALIDATION_ERROR("Invalid input.");
+        if (mNamedInputs != inputs || mNamedOutputs != outputs) {
+            bool anyPointersChanged = false;
+            for (auto& input : inputs->GetRecords()) {
+                if (mExternals.find(input.first) == mExternals.end()) {
```
Comment (fujunwei) [inline src/webnn_native/xnnpack/GraphXNN.cpp:756]: There is a macro named DAWN_INVALID_IF that can simplify the usage.

**Review by fujunwei — APPROVED** [review]
LGTM.

**Review by Honry — APPROVED** [review]
LGTM, thanks!

File: src/webnn_native/xnnpack/GraphXNN.cpp (hunk 1 above)
Comment (huningxin) [reply src/webnn_native/xnnpack/GraphXNN.cpp]: fixed in https://github.com/webmachinelearning/webnn-native/pull/249/commits/1ae8d6501923154a085218b43f45ffbef3e081c8, please take another look. Thanks.

**Comment by fujunwei** [comment]
Merge it.

**Pull request merged**